### DISCUSSION TITLE
Nutzung Tablet/Webserver

### DISCUSSION THREAD

**Discussion by Marvin0795** [AndreWohnsland/CocktailBerry#191] [Ideas]
Könnte man in Zukünftigen Versionen einen Webserver integrieren, mit dem es möglich ist, über ein Endgerät(Tablet) seiner Wahl die Steuerung durchzuführen ? 

Wir planen die Zweite Version mit der Cocktailberry Software, möchten dafür aber gern ein Tablet einsetzen zwecks Platz etc.

**Comment by AndreWohnsland** [comment]
Hi Marvin. Sowas schwebt mir schon länger im Kopf, da die Idee der Benutzung ohne ein direkt verbauten Touchscreen auch nett finde, wird aber mit v2.0 erst kommen. Auch alleine dass der Benutzer nur den Server ansprechen könnte und sein eigene grafische Benutzeroberfläche mitbringt. Wann das ist kann ich schwer sagen, da dafür an sich ein Großteil des Codes umgeschrieben werden muss (wurde damals nicht mit dem Hintergedanken entwickelt) und außerdem ein anderes grafisches Tool verwendet werden muss (aktuelles unterstützt sowas nicht). Ich schätze etwas im guten dreistelligen Zeitbereich, bis eine Beta dafür draußen ist (also dieses Jahr unwahrscheinlich). Sobald es aber eine Testversion davon gibt werde ich dich hier informieren. Ich kann mir gut vorstellen dass es eine Alternative wird, die man zum Beispiel beim installieren angibt und diese dann erstmal in abgespeckter Variante mit dem Haupteigenschaften läuft.
LG

**Reply by AndreWohnsland** [reply]
Wenn ich mir deine Frage nochmal genauer ansehe: willst du dann nur den WS mit gewissen endpoints wie POST cocktail/make/id, GET cocktails/ etc verwenden, und nimmst du eine eigene Oberfläche, oder braucht es auch eine bereitgestellte Benutzeroberfläche vom Server?

**Reply by Marvin0795** [reply]
Hallo Andre, vielen dank für deine schnelle Rückmeldung. 

Ich habe mir schon gedacht, dass hierfür das Programm komplett umgeschrieben werden müsste.

Natürlich wäre es cool, wenn die die Benutzeroberfläche komplett vom Raspberry(Webserver) kommt.
Ich arbeite bei WAGO, hatte früher mal mit Codesys und Webvisu einen versuch mit einer Cocktailmaschine gestartet.
Hierbei liegt der Webserver auch komplett auf der Steuerung.

Eine Weitere Möglichkeit wäre natürlich auch die von dir beschriebene. man kann die Befehle verwenden und baut sich seine eigene GUI.

**Comment by AndreWohnsland** [comment]
So für die Sichtbarkeit, [hier ](https://github.com/users/AndreWohnsland/projects/1) ist das Planning Board für 2.0 (webserver) und [hier](https://github.com/AndreWohnsland/CocktailBerry/milestone/1) der Meilenstein mit den zugehörigen Issues für die Verwirklichung. In den nächsten Wochen sollte die Alpha (Cocktailansicht und Flaschenansicht) verfügbar sein.
